google/cloud/storage/transfer_manager.py

```diff
@@ -1205,6 +1304,38 @@ def _get_pool_class_and_requirements(worker_type):
         )
 
 
+def _digest_ordered_checksum_and_size_pairs(checksum_and_size_pairs):
+    base_crc = None
+    zeroes = bytes(0)
+    for part_crc, size in checksum_and_size_pairs:
+        if not base_crc:
+            base_crc = part_crc
+        else:
+            base_crc ^= 0xFFFFFFFF  # precondition
+
+            # Zero pad base_crc32c. To conserve memory, do so with only
+            # MAX_CRC32C_ZERO_ARRAY_SIZE at a time. Reuse the zeroes array where
+            # possible.
+            padded = 0
+            while padded < size:
+                desired_zeroes_size = min((size - padded), MAX_CRC32C_ZERO_ARRAY_SIZE)
+                # resize zeroes array, unless we already have the correct size.
+                zeroes = (
+                    bytes(desired_zeroes_size)
```

Comment:
what if you did `zeroes[:desired_zeroes_size]`  instead creating a new block?

Reply:
That's a good idea. I'll experiment with it.